docs/book/src/cronjob-tutorial/testdata/finalizer_example.go

```
@@ -67,10 +67,23 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// then lets add the finalizer and update the object. This is equivalent
 		// to registering our finalizer.
 		if !controllerutil.ContainsFinalizer(cronJob, myFinalizerName) {
-			controllerutil.AddFinalizer(cronJob, myFinalizerName)
+			log.Info("Adding Finalizer for CronJob")
+			if ok := controllerutil.AddFinalizer(cronJob, myFinalizerName); !ok {
+				log.Error(err, "Failed to add finalizer into the custom resource")
+				return ctrl.Result{Requeue: true}, nil
```

we could just return the error.